flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointStatistics.java

```diff
@@ -0,0 +1,534 @@
+@JsonSubTypes({
+	@JsonSubTypes.Type(value = CheckpointStatistics.CompletedCheckpointStatistics.class, name = "completed"),
+	@JsonSubTypes.Type(value = CheckpointStatistics.FailedCheckpointStatistics.class, name = "failed")})
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class CheckpointStatistics implements ResponseBody {
+
+	public static final String FIELD_NAME_ID = "id";
+
+	public static final String FIELD_NAME_STATUS = "status";
+
+	public static final String FIELD_NAME_IS_SAVEPOINT = "is_savepoint";
+
+	public static final String FIELD_NAME_TRIGGER_TIMESTAMP = "trigger_timestamp";
+
+	public static final String FIELD_NAME_LATEST_ACK_TIMESTAMP = "latest_ack_timestamp";
+
+	public static final String FIELD_NAME_STATE_SIZE = "state_size";
+
+	public static final String FIELD_NAME_DURATION = "end_to_end_duration";
+
+	public static final String FIELD_NAME_ALIGNMENT_BUFFERED = "alignment_buffered";
+
+	public static final String FIELD_NAME_NUM_SUBTASKS = "num_subtasks";
+
+	public static final String FIELD_NAME_NUM_ACK_SUBTASKS = "num_acknowledged_subtasks";
+
+	public static final String FIELD_NAME_TASKS = "tasks";
+
+	@JsonProperty(FIELD_NAME_ID)
+	private final long id;
+
+	@JsonProperty(FIELD_NAME_STATUS)
+	private final CheckpointStatsStatus status;
+
+	@JsonProperty(FIELD_NAME_IS_SAVEPOINT)
+	private final boolean savepoint;
+
+	@JsonProperty(FIELD_NAME_TRIGGER_TIMESTAMP)
+	private final long triggerTimestamp;
+
+	@JsonProperty(FIELD_NAME_LATEST_ACK_TIMESTAMP)
+	private final long latestAckTimestamp;
+
+	@JsonProperty(FIELD_NAME_STATE_SIZE)
+	private final long stateSize;
+
+	@JsonProperty(FIELD_NAME_DURATION)
+	private final long duration;
+
+	@JsonProperty(FIELD_NAME_ALIGNMENT_BUFFERED)
+	private final long alignmentBuffered;
+
+	@JsonProperty(FIELD_NAME_NUM_SUBTASKS)
+	private final int numSubtasks;
+
+	@JsonProperty(FIELD_NAME_NUM_ACK_SUBTASKS)
+	private final int numAckSubtasks;
+
+	@JsonProperty(FIELD_NAME_TASKS)
+	@JsonSerialize(keyUsing = JobVertexIDSerializer.class)
+	@Nullable
+	private final Map<JobVertexID, TaskCheckpointStatistics> checkpointStatisticsPerTask;
+
+	@JsonCreator
+	private CheckpointStatistics(
+			@JsonProperty(FIELD_NAME_ID) long id,
+			@JsonProperty(FIELD_NAME_STATUS) CheckpointStatsStatus status,
+			@JsonProperty(FIELD_NAME_IS_SAVEPOINT) boolean savepoint,
+			@JsonProperty(FIELD_NAME_TRIGGER_TIMESTAMP) long triggerTimestamp,
+			@JsonProperty(FIELD_NAME_LATEST_ACK_TIMESTAMP) long latestAckTimestamp,
+			@JsonProperty(FIELD_NAME_STATE_SIZE) long stateSize,
+			@JsonProperty(FIELD_NAME_DURATION) long duration,
+			@JsonProperty(FIELD_NAME_ALIGNMENT_BUFFERED) long alignmentBuffered,
+			@JsonProperty(FIELD_NAME_NUM_SUBTASKS) int numSubtasks,
+			@JsonProperty(FIELD_NAME_NUM_ACK_SUBTASKS) int numAckSubtasks,
+			@JsonDeserialize(keyUsing = JobVertexIDDeserializer.class) @JsonProperty(FIELD_NAME_TASKS) @Nullable Map<JobVertexID, TaskCheckpointStatistics> checkpointStatisticsPerTask) {
+		this.id = id;
```

Comment:
missing `checkNotNull` check for all fields but `checkpoitnStatisticsPerTask`.

Reply:
Good catch. Will add the checks.

Reply:
Hmm I think all not checked parameters are actually primitives and, thus, don't need to be checked for null.

Reply:
true, all objects are checked.